.github/workflows/lint-format.yml

```diff
@@ -0,0 +1,35 @@
+name: "Lint & Format Code"
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+permissions:
+  pull-requests: read
+
+jobs:
+  format:
+    name: Format apply
+    timeout-minutes: 20
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - name: Automatically apply lint suggestions
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --fix --workspace
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: Format check packages
+        run: cargo +nightly fmt --all -- --check
+      - name: Commit changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Apply formatting changes
```

Comment:
Can we use conventional commit syntax please - e.g. `style: apply formatting changes`

https://www.conventionalcommits.org/en/v1.0.0/#summary